grid.py

```diff
@@ -611,19 +631,18 @@ def find_last_space_rows(self, pos: int, is_space: bool, first_space: int, last_
         """
         # Black square
         if self._grid[line][pos].letter != None:
-            is_space: bool = False
-            last_space: int = pos - 1
-            if last_space - first_space >= 3:
-                usable_spaces.append((first_space, last_space))
+            self.usable_space.is_space = False
+            self.usable_space.last_space = pos - 1
+            if self.usable_space.last_space - self.usable_space.first_space >= 3:
+                usable_spaces.append((self.usable_space.first_space, self.usable_space.last_space))
         else:
             # Last space in the row, with preceeding spaces
             if pos == self.rows - 1:
-                last_space: int = pos
-                if last_space - first_space >= 3:
-                    usable_spaces.append((first_space, last_space))
-        return is_space
+                self.usable_space.last_space = pos
+                if self.usable_space.last_space - self.usable_space.first_space >= 3:
+                    usable_spaces.append((self.usable_space.first_space, self.usable_space.last_space))
     
-    def find_last_space_cols(self, pos: int, is_space: bool, first_space: int, last_space: int, line: int, usable_spaces: List[List[int]]) -> bool:
+    def find_last_space_cols(self, pos: int, line: int, usable_spaces: List[List[int]]) -> None:
```

Comment:
If the function is being called per column, maybe col instead of cols. Like how we only use plural form for lists, arrays, dict, any collection type